src/Functions/formatDateTime.cpp

```diff
@@ -1172,32 +1386,41 @@ class FunctionFormatDateTimeImpl : public IFunction
     }
 
     template <typename T>
-    size_t parseJodaFormat(const String & format, std::vector<Instruction<T>> & instructions, UInt32, String &) const
+    size_t parseJodaFormat(const String & format, std::vector<Instruction<T>> & instructions, UInt32, bool, String &) const
     {
         /// If the argument was DateTime, add instruction for printing. If it was date, just append default literal
-        auto add_instruction = [&](auto && func [[maybe_unused]], const String & default_literal [[maybe_unused]])
+        auto add_instruction = [&]([[maybe_unused]] typename Instruction<T>::FuncJoda && func, [[maybe_unused]] const String & default_literal)
         {
             if constexpr (std::is_same_v<T, UInt32> || std::is_same_v<T, Int64>)
```

Comment:
maybe move `concept is_any_of` from `typeid_cast.h` to `Common/Concepts.h` and use it here?

Reply:
Sounds like a good idea (didn't know about `is_any_of`), done.